src/common/metrics.rs

```diff
@@ -195,153 +196,147 @@ impl MetricsProvider for RequestsTelemetry {
-
-                if *status == REST_TIMINGS_FOR_STATUS {
-                    avg_secs.push(gauge(
-                        stats.avg_duration_micros.unwrap_or(0.0) as f64 / 1_000_000.0,
-                        &labels,
-                    ));
-                    min_secs.push(gauge(
-                        stats.min_duration_micros.unwrap_or(0.0) as f64 / 1_000_000.0,
-                        &labels,
-                    ));
-                    max_secs.push(gauge(
-                        stats.max_duration_micros.unwrap_or(0.0) as f64 / 1_000_000.0,
-                        &labels,
-                    ));
-                }
+                builder.add(
+                    stats,
+                    &[
+                        ("method", method),
+                        ("endpoint", endpoint),
+                        ("status", &status.to_string()),
+                    ],
+                    *status == REST_TIMINGS_FOR_STATUS,
+                );
             }
         }
-
-        if !total.is_empty() {
-            metrics.push(metric_family(
-                "rest_responses_total",
-                "total number of responses",
-                MetricType::COUNTER,
-                total,
-            ));
-        }
-        if !fail_total.is_empty() {
-            metrics.push(metric_family(
-                "rest_responses_fail_total",
-                "total number of failed responses",
-                MetricType::COUNTER,
-                fail_total,
-            ));
-        }
-        if !avg_secs.is_empty() {
-            metrics.push(metric_family(
-                "rest_responses_avg_duration_seconds",
-                "average response duration",
-                MetricType::GAUGE,
-                avg_secs,
-            ));
-        }
-        if !min_secs.is_empty() {
-            metrics.push(metric_family(
-                "rest_responses_min_duration_seconds",
-                "minimum response duration",
-                MetricType::GAUGE,
-                min_secs,
-            ));
-        }
-        if !max_secs.is_empty() {
-            metrics.push(metric_family(
-                "rest_responses_max_duration_seconds",
-                "maximum response duration",
-                MetricType::GAUGE,
-                max_secs,
-            ));
-        }
+        builder.build("rest", metrics);
     }
 }
 
 impl MetricsProvider for GrpcTelemetry {
     fn add_metrics(&self, metrics: &mut Vec<MetricFamily>) {
-        let (mut total, mut fail_total, mut avg_secs, mut min_secs, mut max_secs) =
-            (vec![], vec![], vec![], vec![], vec![]);
+        let mut builder = OperationDurationMetricsBuilder::default();
         for (endpoint, stats) in &self.responses {
-            // Endpoint must be whitelisted
             if GRPC_ENDPOINT_WHITELIST
                 .binary_search(&endpoint.as_str())
-                .is_err()
+                .is_ok()
```

Comment:
This should be `is_err()`.

If it isn't whitelisted, we `continue` and go to the next item. If the binary search returns an error, it is not whitelisted.

I also recommend to put the comment that we had here back to clarify this.